mineral-ui/mineral-ui
# [Navigation] Add PrimaryNav, SecondaryNav, & NavItem components

### Description
- Prevent disabled non-`button` elements from being focused
- Add `element` prop to MenuItem and allow tabIndex to be overridden
- Export some types from Menu
- Add PrimaryNav and SecondaryNav (made from Navigation), NavOverflowMenu (internal) and NavItem components
- Add examples & demos for PrimaryNav, SecondaryNav, & NavItem
- Add specs & update snapshots

### Motivation and context


Closes #787 

### Screenshots, videos, or demo, if appropriate


https://787-navigation--mineral-ui.netlify.com/components/primary-nav

### How to test


1. `cd mineral-ui && npm start`
1. Navigate to [PrimaryNav](https://787-navigation--mineral-ui.netlify.com/components/primary-nav), [SecondaryNav](https://787-navigation--mineral-ui.netlify.com/components/secondary-nav), & [NavItem](https://787-navigation--mineral-ui.netlify.com/components/nav-item) pages
1. Confirm all looks correct with no typos or errors

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

Bug fix: Button's with a non-button element used to be focusable

### Checklist




### How does this PR make you feel?

![Dog navigating via scooter](https://media.giphy.com/media/l0K4bHSpcFycezcqY/giphy.gif)
